modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMqContainer.java

```diff
@@ -0,0 +1,100 @@
+package org.testcontainers.containers;
+
+import java.time.Duration;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+
+
+/**
+ * This {@link Container} is based on the official RabbitMQ ({@code rabbitmq}) images from
+ * <a href="https://hub.docker.com/_/rabbitmq/">DockerHub</a>.
+ *
+ * @author Stefan Ludwig
+ */
+public class RabbitMqContainer extends GenericContainer<RabbitMqContainer> {
+
+    private static final String DEFAULT_LOG_MESSAGE_REGEX = ".*Server startup complete.*\n";
+    private static final int DEFAULT_STARTUP_TIMEOUT = 30;
+
+    public static final int RABBITMQ_PORT = 5672;
+
+    /**
+     * Creates a new {@link RabbitMqContainer} with the {@code 'rabbitmq:latest'} image.
+     */
+    public RabbitMqContainer() {
+        this("latest");
+    }
+
+    /**
+     * Creates a new {@link RabbitMqContainer} with the {@code 'rabbitmq:${tag}'} image.
+     *
+     * @param tag the tag (e.g. version) to use
+     */
+    public RabbitMqContainer(@NotNull String tag) {
+        super("rabbitmq:" + tag);
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPort(RABBITMQ_PORT);
+        setWaitStrategy(createWaitStrategy());
+    }
+
+    /**
+     * This method can be overridden in order to change the {@link WaitStrategy}
+     * used to determine whether or not the container is ready to be used.
+     * <p>
+     * Per default a {@link LogMessageWaitStrategy} with a pattern provided by
+     * {@link #getContainerReadyLogRegEx()} and a startup timeout provided by
+     * {@link #getStartupTimeout()} is used.
+     *
+     * @return the {@link WaitStrategy} to use
+     */
+    @NotNull
+    protected WaitStrategy createWaitStrategy() {
+        return new LogMessageWaitStrategy() //
```

Comment:
is it really needed? AFAIK MongoDB is ready when the port is ready, and by default Testcontainers will wait for the port

Reply:
Yes, we (in one of my projects) had massive problems until we added the wait on the log message. RabbitMQ is apparently not ready to be used until that point in time, even if the port is open a bit sooner. Depending on the underlying hardware the delay between the container being "ready" without the wait strategy was between 5 and 10 seconds. Only after we included the `WaitStrategy` the container could be used reliably.

Reply:
as for MongoDB (and #803) I'll check that again, I included it there because I had to include it already in RabbitMQ, but maybe MongoDB is a bit more reliable regarding it ports ..

Reply:
maybe it would be good to add a couple of extra tests for that latest major versions of RabbitMQ to make sure my `WaitStrategy` is reliable across them?